ntp2chrony/ntp2chrony.py

```diff
@@ -356,44 +356,29 @@ def get_processed_time_sources(self):
         for source in self.time_sources:
             if source["type"] != "server":
                 continue
-            if not source["address"].endswith(".pool.ntp.org"):
-                continue
             m = re.match("^([0123])(\\.\\w+)?\\.pool\\.ntp\\.org$", source["address"])
             if m is None:
                 continue
             number = m.group(1)
             zone = m.group(2)
             if zone not in pools:
                 pools[zone] = []
-            pools[zone].append((number, source))
+            pools[zone].append((int(number), source))
 
         remove_servers = set()
-        convert_servers = set()
         for zone, pool in pools.items():
+            # sort and skip all pools not in [0, 3] range
             pool.sort()
-            if [number for number, source in pool] != ["0", "1", "2", "3"]:
+            if [number for number, source in pool] != [0, 1, 2, 3]:
+                # only exact group of 4 servers can be converted, nothing to do here
                 continue
-            for i in range(1, len(pool)):
-                if pool[0][1]["options"] != pool[i][1]["options"]:
-                    break
-            else:
-                assert len(pool) > 2
-                for i in range(len(pool)):
-                    if i == 2:
-                        convert_servers.add(pool[i][1]["address"])
-                    else:
-                        remove_servers.add(pool[i][1]["address"])
-
-        processed_sources = []
-        for source in self.time_sources:
-            if source["type"] == "server":
-                if source["address"] in remove_servers:
-                    continue
-                if source["address"] in convert_servers:
-                    source["type"] = "pool"
-            processed_sources.append(source)
+            # verify that parameters are the same for all servers in the pool
+            if not all([p[1]["options"] == pool[0][1]["options"] for p in pool]):
+                break
+            remove_servers.update([pool[i][1]["address"] for i in [0, 1, 3]])
+            pool[2][1]["type"] = "pool"
 
-        return processed_sources
+        return [s for s in self.time_sources if s["address"] not in remove_servers]
```

Comment:
I think this should remove only server sources, not peers or pools.

Reply:
Sorry not getting you, why are you saying that peers or pools are removed? remove_servers contains a set of server sources 0, 1, 3 only.
The final list comprehension is equivalent to the original code
```
        processed_sources = []
        for source in self.time_sources:
            if source["type"] == "server":
                if source["address"] in remove_servers:
                    continue
                if source["address"] in convert_servers:
                    source["type"] = "pool"
            processed_sources.append(source)
```

Reply:
There is no check of the type. It's not equivalent. You can try it with this config:
```
server 0.pool.ntp.org
server 1.pool.ntp.org
server 2.pool.ntp.org
server 3.pool.ntp.org
peer 0.pool.ntp.org
```

Reply:
Understood, thanks, added this as another test case to fixate.